### ISSUE TITLE
use files in sketch folder to set the sketch's window icon

### ISSUE THREAD

**Issue by benfry** [processing/processing#5225]
To address https://github.com/processing/processing/issues/5123 and https://github.com/processing/processing/pull/5202, we need an implementation that will handling custom icons to get around our issues with differences in `surface` between OpenGL and the default renderer. 

[This comment](https://github.com/processing/processing/issues/5123#issuecomment-309870870) describes the approach we'll be using to fix it once and for all.

**Comment by codeanticode** [comment]
@benfry is this something you are planning to implement for the next release? Need any help?

**Comment by benfry** [comment]
Yeah, planning to. Didn't include it in the last release because I didn't want to mix feature changes with bug fixes. I'll have you take a crack at the OpenGL part once I have the API in place.

**Comment by codeanticode** [comment]
Ok, sounds good!

**Comment by cukiakimani** [comment]
@benfry Do we have an update on this?
